### PR TITLE
Added getInRange() to ScheduledInterviewDao 

### PR DESCRIPTION
**Pull request by apluscs**
Purpose: to get all Scheduled Interviews for a shadow to choose from, later we can tweak the method to filter by a specific job
Side note: Everything related to Shadow will go into the shadow branch, so master doesn't get messed up in prep for the demo

### CODE REVIEW

**Review by GrantComm — APPROVED** [review]

File: src/main/java/com/google/sps/data/DatastoreScheduledInterviewDao.java [hunk 1]
```diff
@@ -60,6 +60,19 @@ public DatastoreScheduledInterviewDao() {
     return Optional.of(entityToScheduledInterview(scheduledInterviewEntity));
   }
 
+  /**
+   * Returns a list of all scheduled ScheduledInterview objects whose startTime is between minTime
```
Comment (meagancheese) [inline src/main/java/com/google/sps/data/DatastoreScheduledInterviewDao.java]: No need to say scheduled ScheduledInterview. minTime and maxTime are not in milliseconds, and just say 'objects between...', don't bring startTime into it.

File: src/main/java/com/google/sps/data/FakeScheduledInterviewDao.java [hunk 2]
```diff
@@ -58,6 +58,22 @@ public FakeScheduledInterviewDao() {
     return Optional.empty();
   }
 
+  /**
+   * Returns a list of all scheduled ScheduledInterview objects whose startTime is between minTime
+   * and maxTime. minTime and maxTime are in milliseconds.
```
Comment (meagancheese) [inline src/main/java/com/google/sps/data/FakeScheduledInterviewDao.java]: Fix this comment similarly to the other one.

File: src/main/java/com/google/sps/data/DatastoreScheduledInterviewDao.java (hunk 1 above)
Comment (meagancheese) [inline src/main/java/com/google/sps/data/DatastoreScheduledInterviewDao.java]: Are they sorted?

Comment (meagancheese) [reply src/main/java/com/google/sps/data/DatastoreScheduledInterviewDao.java]: I think they are, say as much.

File: src/main/java/com/google/sps/data/FakeScheduledInterviewDao.java (hunk 2 above)
Comment (meagancheese) [reply src/main/java/com/google/sps/data/FakeScheduledInterviewDao.java]: Also, be sure to sort here.

File: src/main/java/com/google/sps/data/ScheduledInterviewDao.java [hunk 3]
```diff
@@ -26,11 +26,16 @@
   // Returns the ScheduledInterview object.
   public Optional<ScheduledInterview> get(long id);
 
-  // Returns a list of the ScheduledInterview objects that the user participates in.
+  // Returns a list, sorted by start time, of all scheduled ScheduledInterview objects ranging from
+  // minTime to maxTime. minTime and maxTime are in milliseconds.
```
Comment (meagancheese) [inline src/main/java/com/google/sps/data/ScheduledInterviewDao.java]: minTime and maxTime are not in milliseconds, they're Instants.

File: src/main/java/com/google/sps/data/ScheduledInterviewDao.java [hunk 4]
```diff
@@ -26,11 +26,16 @@
...
+  public List<ScheduledInterview> getInRange(Instant minTime, Instant maxTime);
+
+  // Returns a list, sorted by start time, of the ScheduledInterview objects that the user
+  // participates in.
   public List<ScheduledInterview> getForPerson(String email);
 
-  // Returns a list of all scheduled ScheduledInterview objects ranging from minTim to maxTime of a
-  // user. minTime and maxTime are in milliseconds.
+  // Returns a list, sorted by start time, of all scheduled ScheduledInterview objects ranging from
+  // minTime to maxTime of a user. minTime and maxTime are in milliseconds.
```
Comment (meagancheese) [inline src/main/java/com/google/sps/data/ScheduledInterviewDao.java]: minTime and maxTime are not in milliseconds.

**Review by meagancheese — CHANGES_REQUESTED** [review]
Mostly comment changes, and then making sure to sort in Fake Dao

File: src/main/java/com/google/sps/data/ScheduledInterviewDao.java [hunk 5]
```diff
@@ -26,10 +26,14 @@
   // Returns the ScheduledInterview object.
   public Optional<ScheduledInterview> get(long id);
 
+  // Returns a list, sorted by start time, of all scheduled ScheduledInterview objects ranging from
```
Comment (meagancheese) [inline src/main/java/com/google/sps/data/ScheduledInterviewDao.java]: Nit: Remove the word "scheduled"

File: src/main/java/com/google/sps/data/ScheduledInterviewDao.java [hunk 6]
```diff
@@ -26,10 +26,14 @@
...
+  // minTime to maxTime.
+  public List<ScheduledInterview> getInRange(Instant minTime, Instant maxTime);
+
   // Returns a list of the ScheduledInterview objects that the user participates in.
   public List<ScheduledInterview> getForPerson(String userId);
 
-  // Returns a list of all scheduled ScheduledInterview objects ranging from minTim to maxTime of a
+  // Returns a list of all scheduled ScheduledInterview objects ranging from minTime to maxTime of a
```
Comment (meagancheese) [inline src/main/java/com/google/sps/data/ScheduledInterviewDao.java]: Nit: remove the word "scheduled"

File: src/test/java/com/google/sps/data/DatastoreScheduledInterviewTest.java [hunk 7]
```diff
@@ -238,4 +238,62 @@ public void getsScheduledInterviewsInRange() {
 
     Assert.assertEquals(expected, result);
   }
+
+  // Tests retrieving all scheduledInterviews from Datastore, non-empty result.
+  @Test
+  public void getsScheduledInterviewInRange() {
```
Comment (meagancheese) [inline src/test/java/com/google/sps/data/DatastoreScheduledInterviewTest.java]: Nit: Interviews (plural)

File: src/test/java/com/google/sps/data/DatastoreScheduledInterviewTest.java [hunk 8]
```diff
@@ -238,4 +238,62 @@ public void getsScheduledInterviewsInRange() {
...
+    expected.add(expectedScheduledInterview1);
+    expected.add(expectedScheduledInterview2);
+    expected.add(expectedScheduledInterview3);
+
+    Assert.assertEquals(expected, result);
+  }
+
+  // Tests retrieving all scheduledInterviews from Datastore, empty result.
+  @Test
+  public void getsScheduledInterviewInRangeEmpty() {
```
Comment (meagancheese) [inline src/test/java/com/google/sps/data/DatastoreScheduledInterviewTest.java]: Nit: again, plural

**Review by meagancheese — APPROVED** [review]
A handful of nits, otherwise LGTM.

**Review by jalexanderqed — APPROVED** [review]

**Pull request merged**